docs/05-administrator-guide/02-configuration/04-performance/02-topology.md

```diff
@@ -4,4 +4,197 @@ sidebar_label: Topology
+
+  ```xml
+  <property>
+    <name>net.topology.node.switch.mapping.impl</name>
+    <value>org.apache.hadoop.net.ScriptBasedMapping</value>
+  </property>
+  <property>
+    <name>net.topology.script.file.name</name>
+    <value>/etc/ozone/determine_rack.sh</value>
+  </property>
+  ```
+
+- **Script:** Admin-provided, executable script. Ozone passes IPs (up to `net.topology.script.number.args`, default 100) as arguments; script outputs rack paths (one per line).
+  Example `determine_rack.sh`:
+
+  ```bash
+  #!/bin/bash
+  # This is a simplified example. A real script might query a CMDB or use other logic.
+  while [ $# -gt 0 ] ; do
+    nodeAddress=$1
+    if [[ "$nodeAddress" == "192.168.1.100" || "$nodeAddress" == "datanode101.example.com" ]]; then
+      echo "/rack1"
+    elif [[ "$nodeAddress" == "192.168.1.102" || "$nodeAddress" == "datanode103.example.com" ]]; then
+      echo "/rack2"
+    else
+      echo "/default-rack"
+    fi
+    shift
+  done
+  ```
+
+Ensure the script is executable (`chmod +x /etc/ozone/determine_rack.sh`).
+
+**Note:** For production environments, implement robust error handling and validation in your script. This should include handling network timeouts, invalid inputs, CMDB query failures, and logging errors appropriately. The example above is simplified for illustration purposes only.
+
+**Topology Mapping Best Practices:**
+
+- **Accuracy:** Mappings must be accurate and current.
+- **Static Mapping:** Simpler for small, stable clusters; requires manual updates.
+- **Dynamic Mapping:** Flexible for large/dynamic clusters. Script performance, correctness, and reliability are vital; ensure it's idempotent and handles batch lookups efficiently.
+
+## Placement and Selection Policies
+
+Ozone uses three distinct types of policies to manage how and where data is written.
+
+### 1. Pipeline Creation Policy
+
+This policy selects a set of Datanodes to form a new pipeline. Its purpose is to ensure new pipelines are internally fault-tolerant by spreading their nodes across racks, while also balancing the number of pipelines across the Datanodes. This is the primary mechanism for topology awareness on the write path for open containers.
+
+The policy is configured by the `ozone.scm.pipeline.placement.impl` property in `ozone-site.xml`.
+
+- **`PipelinePlacementPolicy` (Default)**
+
+  - **Function:** This is the default and only supported policy for pipeline creation. It chooses Datanodes based on load balancing (pipeline count per node) and network topology. It filters out nodes that are too heavily engaged in other pipelines and then selects nodes to ensure rack diversity. This policy is recommended for most production environments.
+  - **Use Cases:** General purpose pipeline creation in a rack-aware cluster.
+
+### 2. Pipeline Selection (Load Balancing) Policy
```

Comment:
We should make it clear that the section 2 and section 3 are for Ratis (replicated) containers. Perhaps even refactor them into a sub-page.